raddb/mods-available/linelog

```diff
@@ -34,22 +49,30 @@ linelog {
 	#  If the configuration entry cannot be found, then no log
 	#  message is printed.
 	#
-	#  i.e. You can have many log messages in one "linelog" module.
+	#  i.e: You can have many log messages in one `linelog` module.
 	#  If this two-step expansion did not exist, you would have
-	#  needed to configure one "linelog" module for each log message.
+	#  needed to configure one `linelog` module for each log message.
+	#  ====
+	#
 
 	#
-	#  Reference the Packet-Type (Access-Request, etc.)  If it doesn't
-	#  exist, reference the "format" entry, above.
+	#  reference::
+	#
+	#  Reference the `Packet-Type` (`Access-Request`, etc.)  If it doesn't
+	#  exist, reference the `format` entry, above.
+	#
+	#  May be an attribute ref, `xlat`, `literal` or `exec`.
```

Comment:
I guess we don't have these docs merged in yet.  Maybe talk to @alandekok  and see if we can get the NR docs included in the doc source tree.

Reply:
That'd also be `attribute ref` in backticks ;)

Reply:
How about:

```
May be an attribute reference, e.g. `Usert-Name`, or `xlat`, `literal` or `exec`.
```

Reply:
An attribute reference is at the same level as xlat, literal or exec.  We could shorten it to `attref`? It'd be `&User-Name` if you wanted to give an example.

Reply:
For @jpereira 's benefit.

- attribute reference is a description of an attribute and where to find it. e.g. `&reply:Reply-Message` it all supports all the weird and wonderful accessor syntaxes like `&reply:Reply-Message[5]` to get the sixth instance of the reply-message in the reply list.
- xlat is a string expansion, it can call a function like `%{sql:<query>}` or be used to interpolate attribute values, in which case the syntax is the same as attribute references but without the leading & e.g. %{reply:Reply-Message[5]}.
- Literal is a string literal e.g. 'I am a string literal'.  Depending on the quotes used, different escape sequences are supported. 
- Exec is a call out to a script. The script is places in backticks `/usr/bin/myscript.sh -user %{User-Name}`

Reply:
It's probably worth adding another page that explains all of this, and then referencing that page from this (and other) locations.

The alternative is to have the same text repeated everywhere, which makes me unhappy.

Reply:
I'm not convinced all those backticks above are correct. Backticks are generally to show code blocks or examples of something that's being typed, or will appear on screen. So "The `xlat` is set here" would be wrong ("xlat" is just what it's called), but "set the `reference` here" would be correct (as above) _if_ there is an actual option called "reference" that is being set (the option is something that appears on the screen).

So I would say that the last line should lose all its backticks - it's simply

    May be an attribute reference, e.g. `&User-Name`, an xlat, a string literal or an exec.

Better (as that sentence confused me too for a while) might be to make them all links to other pages that explain what they are, or have examples.

    May be
      - an attribute reference, e.g. `&User-Name`;
      - an xlat, e.g. `"%{sql:...}"`;
      - a string literal, such as `"a literal string"`;
      - or an exec, e.g. `` `/bin/echo hello` ``

As @alandekok said, duplication is bad. Small examples might be OK, but the main explanation  of what these are should be on another page and be linked to.

Reply:
@mcnewton The backticks here is following the asciidoc https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/ manual and here in special has a different behavior because the script https://github.com/FreeRADIUS/freeradius-server/blob/master/scripts/asciidoc/conf2adoc does some extra parsing.

Reply:
About the duplication, I agree that we should avoid to duplicate. An option could be to create a kind of glossary with the usual options/sections e.g: pool { ... }

Reply:
A common glossary is useful.  Maybe `doc/raddb/common/` ?

We can put manually generated files there, and refer to them from the config items.

Though in this case, it's really more common *concepts*, and could go into a "reference" documentation

Reply:
The asciidoc manual explains what the markup does, but I don't see any more than it says that backticks make something monospace.

So my question is, why would you write "This can be a `string literal`" rather than "This can be a string literal"?

I don't see a reason for "string literal" to be monospaced, and equivalently no need for "xlat" or "exec" to be either. The _contents_ of them, yes, such as "`"this text"` is a string literal" or "`%{sql:...}` is an xlat", but not the name itself.

The other ones are all good, though :)

I'm not sure what you mean by the conf2adoc script. It would convert e.g. `` `exec` `` to a link to the module of the same name, but in this case "exec" is nothing to do with that module.

Reply:
> A common glossary is useful. Maybe `doc/raddb/common/` ?

That sounds good - tidying up `doc/` would be nice, and then linking to more full documentation there where it's not directly part of the config makes sense.

In theory it doesn't even need to be `doc/raddb/...`, just `doc/glossary.md` or `doc/guide/eap.md` for example, as by the fact they're being moved out of `raddb/` means they're more general documentation.